### PR TITLE
Custom Transport Configuration support

### LINKED ISSUE #372: Add ability to configure default response status code for ApacheHttpClientTransaport in performCall
### Environment

* Checkout SDK version: 6.0.26
* Platform and version: Java 8
* Operating System and version: Amazon Linux 2

### Description
In the ApacheHttpClientTransport class of the Checkout SDK, in the [performCall method](https://github.com/checkout/checkout-sdk-java/blob/6.0.26/src/main/java/com/checkout/ApacheHttpClientTransport.java#L178), the default catch-all status code for the response is a bad request (400); this is thrown for a SocketTimeoutException for example. We want this status code to be customizable with the ability to provide an alternate status code, such as an internal server error (500) or gateway timeout (504).

## Proposed Solution
Could you provide a way to customize this with the configuration object, with the default being the existing bad request and the ability to pass a custom status code?

- [x] I may be able to implement this feature

### PR DESCRIPTION
**Pull request by dlabey**
- Adds the concept of TransportConfiguration for
- Adds support for setting a custom HTTP status code in the performCall method or use the same default bad request

Fixes #372

### CODE REVIEW

File: src/main/java/com/checkout/CustomTransportConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.checkout;
+
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class CustomTransportConfiguration implements TransportConfiguration {
+
+    private int defaultHttpStatusCode;
```
[the remark this answers is not in the document]
Comment (armando-rodriguez-cko) [reply src/main/java/com/checkout/CustomTransportConfiguration.java:10]: Please @dlabey add Override annotation here

File: src/main/java/com/checkout/DefaultTransportConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.checkout;
+
+import org.apache.http.HttpStatus;
+
+import lombok.Getter;
+
+@Getter
+public class DefaultTransportConfiguration implements TransportConfiguration {
+
+    private final int defaultHttpStatusCode;
```
[the remark this answers is not in the document]
Comment (armando-rodriguez-cko) [reply src/main/java/com/checkout/DefaultTransportConfiguration.java:10]: Please @dlabey add Override annotation here

**Comment by armando-rodriguez-cko** [comment]
Please @dlabey review all the tests, some test are failing

**Comment by dlabey** [comment]
@armando-rodriguez-cko because I am on a fork I am not inheriting the IT secrets from the GitHub environment, and the tests have hardcoded processing channels... how do you advise I proceed?

**Comment by dlabey** [comment]
@armando-rodriguez-cko I changed the build-pull-request file to use pull_request_target so it inherits the context of the source repo. According to https://blog.gitguardian.com/github-actions-security-cheat-sheet/, this can be insecure unless you manually approve the execution of workflows, which you are doing, so maybe this will help.

**Comment by dlabey** [comment]
@armando-rodriguez-cko I have merged it back, could you checkout this branch and run it with the secrets locally? Or could you take this branch and put it into the checkout owned repo for a PR to inherit the secrets?

**Comment by dlabey** [comment]
@armando-rodriguez-cko any update? I cannot run the IT properly because my fork does not inherit the IT Checkout API keys which are coupled to the processing channels used in the IT tests...

**Comment by armando-rodriguez-cko** [comment]
Hi @dlabey, we are working on it, sorry by the delay, we will try to finish it today.

**Review by armando-rodriguez-cko — APPROVED** [review]

**Review by vitor-carvalho-cko — APPROVED** [review]

**Pull request merged**